app/controllers/projects_controller.rb

```diff
@@ -129,6 +129,10 @@ def show_tree_content
 
   def show
     @images = []
+    if @project.blank?
+    redirect_to dashboard_path
```

Comment:
I was also wondering whether to render 404 or redirect to dashboard. But, now it clear, thanks :)
I'll fix the code indentation and if statement.
hot_deploy shouldn't be there. No worries, I'll fix that too if it there.